### ISSUE TITLE
Cannot bind dns interface to port 53 along with systemd-resolved

### ISSUE THREAD

**Issue by gionn** [hashicorp/consul#4578]
Hi, I was trying to setup consul 1.2.2 on docker (with network 'host') on an ubuntu 18.04 node.

I followed what is documented [here](https://www.consul.io/docs/guides/forwarding.html#systemd-resolved-setup), but I am stuck with:

```
==> Starting Consul agent...
==> Error starting agent: agent: timeout starting DNS servers
```

even if I am starting consul on a specific bind address (and port 53):

```
agent -server -ui -config-dir /etc/consul.d -bind 10.240.0.3 -client 10.240.0.3 -dns-port 53 -bootstrap -recursor 8.8.8.8
```

I think the reason is related to systemd-resolved that is binding on 127.0.0.53:53 but it should be definitively possible that consul can bind the same port on a different address.

**Comment by gionn** [comment]
After more investigation, I found that the problem was the same described in https://github.com/hashicorp/consul/issues/3848#issuecomment-361704991, and a solution is described in https://hub.docker.com/_/consul/ (Exposing Consul's DNS Server on Port 53).

Maybe a notice on the documentation could save some time to many other people.

Cheers.

**Comment by mkeeler** [comment]
@gionn After your investigation would you say that the documentation is lacking a notice that using port 53 will require root privileges (or at least running with the CAP_NET_BIND_SERVICE capability). It wasn't an issue with systemd-resolved being bound to 127.0.0.53:53 correct?

**Comment by gionn** [comment]
There is the `CONSUL_ALLOW_PRIVILEGED_PORTS=` environment variable but is documented only on the docker image README and not referenced elsewhere. It could be great to have that somewhere on the official docs, I think that docker is one of the most common deployment scenario.

Systemd-resolved conflicts arises only when consul tried to bind client interfaces to 0.0.0.0 (the default?), with docker I usually configure the docker bridge ip of the VM.

The worst thing for me is that consul is not returning a specific error message for every bind error, that makes troubleshooting very hard.

Thanks.
